src/main.cpp

```diff
@@ -2956,13 +2976,27 @@ bool Session::SeekTime(double seekTime, unsigned int streamId, bool preceeding)
         (streamId == 0 || (*b)->info_.GetPhysicalIndex() == streamId))
     {
       bool bReset;
+      // all streams must be started before seeking to ensure cross chapter seeks
+      // will seek to the correct location/segment
+      if (!(*b)->reader_->IsStarted())
+      {
+        (*b)->stream_.seek_time(
+            static_cast<double>(seekTimeCorrected - ptsDiff) / STREAM_TIME_BASE,
+            preceeding, bReset);
+        if (bReset)
+          (*b)->reader_->Reset(false);
+        bool bStarted;
+        (*b)->reader_->Start(bStarted);
```

Comment:
This makes no sense and is bad code. `bStarted` could have any value as it's a basic type. It should be initialised. Plus it's copied in the call to start so is not used at all!

Reply:
Ok, so I will make sure to initialise bStarted. I'm using it based off what happens here:
https://github.com/xbmc/inputstream.adaptive/blob/1247d4bcff218db54c3c2ec30be020913731f09d/src/main.cpp#L2870-L2877

The `bStarted` bool parameter inside `Start` is by reference so I can't send in a temporary variable (much like `bReset` above) but on looking at the snippet above perhaps I should be checking `bStarted` afterwards and setting `changed_`. And also could save on code duplication by splitting this off as a function.
Does that sound good?

Reply:
Yes, if you don’t do anything with the value I’m not sure there is a point in setting it.

Reply:
All sorted (I think!)